pkg/planner/cardinality/row_count_column.go

```diff
@@ -177,7 +177,7 @@ func equalRowCountOnColumn(sctx planctx.PlanContext, c *statistics.Column, val t
 		if modifyCount == 0 {
 			return 0, nil
 		}
-		return 1, nil
+		return max(1, float64(modifyCount)/float64(c.Histogram.NDV)), nil
```

Comment:
Shouldn't we be writing any test case for this ?

Reply:
A test using real modifyCount will unfortunately be flaky - because the updating of modifyCount after modifications to a table is a timing issue.